activesupport/lib/active_support/actionable_error.rb

```diff
@@ -9,12 +9,99 @@ module ActiveSupport
   module ActionableError
     extend Concern
 
-    class NonActionable < StandardError; end
+    class NonActionable < StandardError; end # :nodoc:
+
+    Trigger = Struct.new(:actionable, :condition) do # :nodoc:
+      def act_on(error)
+        raise actionable if condition&.call(error)
```

Comment:
```suggestion
        raise actionable if condition.call(error)
```

Condition is required. It'll never be nil.